File: Platform/mo-2-core/src/core-java/org/mo/core/aop/config/XAopDefine.java
package org.mo.core.aop.config;

import org.mo.com.lang.RString;
import org.mo.com.validator.RStringValidator;
import org.mo.com.xml.FXmlNode;

//============================================================
// <T>AOP设置定义。</T>
//============================================================
public class XAopDefine
      extends XAbsAopNode
{
   // 标签名称
   public static final String TAG_NAME = "Define";

   // 属性名称
   public static final String PTY_NAME = "name";

   // 名称
   private String _name;

   // 内容
   private String _value;

   // 条件
   private String _condition;

   //============================================================
   // <T>构造AOP设置定义。</T>
   //============================================================
   public XAopDefine(){
   }

   //============================================================
   // <T>获得主键。</T>
   //
   // @return 主键
   //============================================================
   @Override
   public String key(){
      return _name;
   }

   //============================================================
   // <T>获得名称。</T>
   //
   // @return 名称
   //============================================================
   public String name(){
      return _name;
   }

   //============================================================
   // <T>获得内容。</T>
   //
   // @return 内容
   //============================================================
   public String value(){
      return _value;
   }

   //============================================================
   // <T>获得条件。</T>
   //
   // @return 条件
   //============================================================
   public String condition(){
      return _condition;
   }

   //============================================================
   // <T>加载配置节点信息。</T>
   //
   // @param xconfig 配置节点
   //============================================================
   @Override
   public void loadConfig(FXmlNode config){
      super.loadConfig(config);
      // 获得名称
      _name = RString.trim(config.get(PTY_NAME));
      RStringValidator.checkEmpty(_name, PTY_NAME);
      // 获得内容
      _value = RString.trim(config.text());
      // 获得条件
      _condition = RString.trim(config.getNvl("condition"));
   }
}

File: Tools/1 - Common/MoCore/Aop/Config/XAopDefine.cs
using MO.Common.Content;
using MO.Common.Lang;
using MO.Common.System;

namespace MO.Core.Aop.Config {

   public class XAopDefine : XAopNode {

      private static ILogger _logger = RLogger.Find(typeof(XAopDefine));

      public const string TAG = "Define";

      public const string PTY_NAME = "name";

      public const string PTY_VALUE = "value";

      public const string PTY_FORMATOR = "formator";

      private string _name;

      private string _value;

      public override string Id {
         get { return _name; }
      }

      public string Name {
         get { return _name; }
         set { _name = value; }
      }

      public string Value {
         get { return _value; }
         set { _value = value; }
      }

      public XAopDefine() {
      }

      public override void LoadConfig(FXmlNode config) {
         _name = config[PTY_NAME];
         RString.CheckEmpty(_name, "name");
         _value = config[PTY_VALUE];
         RString.CheckEmpty(_value, "value");
         // Load formator
         if (config.Contains(PTY_FORMATOR)) {
            /*string formator = config[PTY_FORMATOR];
            _formator = (IFormator)RClass.CreateInstance(formator);
            _formator.LoadConfig(config);
            // Load formator value
            FFormatorParameters parameters = new FFormatorParameters();
            parameters.RaiseException = true;
            parameters.Value = Value;
            if (_formator.Format(parameters)) {
               _value = RString.Nvl(parameters.Value);
            }*/
         }
         // Dump
         if (_logger.DebugAble) {
            _logger.Debug(this, "LoadConfig", "Define {0}=[{1}]", _name, _value);
         }
      }

   }

}
